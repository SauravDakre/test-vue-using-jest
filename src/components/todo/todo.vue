<template>
    <div>
        <h2>Todo Component</h2>
        <input type="text" v-model='newTask'>
        <button v-on:click="add">add</button>
        <ul>
            <li v-for="task in tasks">{{task}}</li>
        </ul>
    </div>
</template>
<script>
    export default{
        name: 'todo',
        data(){
            return{
                tasks: [],
                newTask: ''
            }
        },
        methods: {
            add: function(){
                this.tasks.push(this.newTask);
                this.newTask = '';
            }
        }
    }
</script>