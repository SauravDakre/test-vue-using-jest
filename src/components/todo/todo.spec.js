import {shallow} from '@vue/test-utils';
import { mount } from '@vue/test-utils';
import todo from './todo.vue';
// import sinon from 'sinon'
let todoComponent;
let todoUi;

describe('todo.spec.js - unit test', function(){
    beforeEach(function(){
        todoComponent = shallow(todo)
    })

    it('initially tasks should be empty', function(){
        expect(todoComponent.vm.tasks.length).toBe(0);
    })

    it('new task is properly populated', function(){
        todoComponent.vm.newTask = 'task1';
        todoComponent.vm.add();
        expect(todoComponent.vm.tasks.length).toBe(1);
        expect(todoComponent.vm.tasks[0]).toEqual('task1');
        expect(todoComponent.vm.newTask).toEqual('')
    })
})

describe('todo.spec.js - integration test', function(){
    beforeEach(function(){
        todoUi = mount(todo);
    })

    it('should not have any list item', function(){
        expect(todoUi.contains('li')).toBe(false)
    })

    it('should have heading', function(){
        const heading = todoUi.find('h2');
        expect(heading.text()).toEqual('Todo Component')
    })

    it('should add new task', function(){
        const input = todoUi.find('input');
        input.element.value = 'task1';
        input.trigger('input')
        todoUi.find('button').trigger('click');
        expect(todoUi.contains('li')).toBe(true)
        expect(todoComponent.vm.tasks[0]).toEqual('task1');
        // console.log(todoComponent.vm.tasks)
    })
})