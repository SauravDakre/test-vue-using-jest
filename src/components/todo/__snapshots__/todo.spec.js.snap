// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo.spec.js - integration test should match snapshot 1`] = `
<div>
  <h2>
    Todo Component
  </h2>
   
  <input
    type="text"
  />
   
  <button>
    add
  </button>
   
  <ul>
    <li>
      task1
    </li>
  </ul>
</div>
`;
